Improved Explanations and Code Comments in Modules

emhgit opened the pull request:
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

Several modules in `/2_Bronze` contain Java and Python code blocks that either lack inline comments or do not provide clear, beginner-friendly explanations. This makes it harder for new learners to understand the code, especially if they are not familiar with the language. I added comments and explanations in the affected files to improve unclear code/explanations.

SansPapyrus683 commented on content/2_Bronze/Intro_DS.mdx:
```diff
@@ -647,6 +647,10 @@ public class PairDemo {
 	}
 }
 
+/* The generic types K and V are used to allow the data types to be specified by the
+ * caller, bypassing the usual static typing of Java
+ */
```
```cpp
/*
 * format multiline comments
 * like this
 */
```

SansPapyrus683 commented on content/2_Bronze/Intro_Greedy.mdx:
```diff
@@ -85,17 +85,22 @@ int main() {
 	cin >> A >> B;
 	ll ans = 0;
-	bool mismatched = false;
+	bool mismatched = false;  // Tracks if we are currently in a mismatched segment
```
this one seems obvious

SansPapyrus683 commented on content/2_Bronze/Intro_Greedy.mdx:
```diff
@@ -85,17 +85,22 @@ int main() {
 			mismatched = false;
 		}
 	}
+	// Output the number of mismatched segments, which is the minimum number of flips
+	// needed
```
is this one needed?

SansPapyrus683 commented on content/2_Bronze/Intro_Greedy.mdx:
```diff
@@ -118,17 +123,22 @@ public class Solution {
 		char[] a = io.next().toCharArray();
 		char[] b = io.next().toCharArray();
 		int ret = 0;
+		// Repeat until both strings are equal
 		while (!new String(a).equals(new String(b))) {
```
oh, are the c++/java code not consistent in terms of code structure?
if so give me some time and i'll rewrite it

emhgit replied:
yes, some of the implementations aren’t consistent; for the code comments that seem very obvious, you can remove them

SansPapyrus683 reviewed (changes requested):
just first couple of files

SansPapyrus683 commented:
alright i sansified the greedy module, think you could check if the comments i added are good enough?

emhgit commented:
> alright i sansified the greedy module, think you could check if the comments i added are good enough?



> alright i sansified the greedy module, think you could check if the comments i added are good enough?

yep, lgtm 👍

SansPapyrus683 commented:
alright i removed the comment regarding generics- we can probably assume basic ap cs knowledge

SansPapyrus683 commented on content/2_Bronze/Intro_Sets.mdx:
````diff
@@ -933,11 +952,13 @@ int main() {
 	}
 }
 ```
+*We use `std::map` because it can handle very large keys efficiently and only stores keys that are set. Accessing a key not present in the map returns `0` by default for integer values.*
````
i mean like what else would we use here lol
